tests/unit/test_detect_target_brightspot.py

```diff
@@ -0,0 +1,172 @@
+from modules.common.modules.logger import logger
+
+
+NUMBER_OF_IMAGES = 7
+TEST_PATH = pathlib.Path("tests", "brightspot_example")
+IMAGE_FILES = [f"ir{i}.png" for i in range(1, NUMBER_OF_IMAGES + 1)]
+EXPECTED_DETECTIONS_PATHS = [
+    pathlib.Path(TEST_PATH, f"bounding_box_ir{i}.txt") for i in range(1, NUMBER_OF_IMAGES + 1)
+]
+TEST_CASES = list(zip(IMAGE_FILES, EXPECTED_DETECTIONS_PATHS))
+
+
+BOUNDING_BOX_PRECISION_TOLERANCE = 3
+CONFIDENCE_PRECISION_TOLERANCE = 6
+
+
+# Test functions use test fixture signature names and access class privates
+# No enable
+# pylint: disable=protected-access,redefined-outer-name,duplicate-code
+
+
+def compare_detections(
+    actual: detections_and_time.DetectionsAndTime, expected: detections_and_time.DetectionsAndTime
+) -> None:
+    """
+    Compare expected and actual detections.
+    """
+    assert len(actual.detections) == len(expected.detections)
+
+    for actual_detection, expected_detection in zip(actual.detections, expected.detections):
+        assert expected_detection.label == actual_detection.label
+
+        np.testing.assert_almost_equal(
+            actual_detection.confidence,
+            expected_detection.confidence,
+            decimal=CONFIDENCE_PRECISION_TOLERANCE,
+        )
+
+        np.testing.assert_almost_equal(
+            actual_detection.x_1,
+            expected_detection.x_1,
+            decimal=BOUNDING_BOX_PRECISION_TOLERANCE,
+        )
+
+        np.testing.assert_almost_equal(
+            actual_detection.y_1,
+            expected_detection.y_1,
+            decimal=BOUNDING_BOX_PRECISION_TOLERANCE,
+        )
+
+        np.testing.assert_almost_equal(
+            actual_detection.x_2,
+            expected_detection.x_2,
+            decimal=BOUNDING_BOX_PRECISION_TOLERANCE,
+        )
+
+        np.testing.assert_almost_equal(
+            actual_detection.y_2,
+            expected_detection.y_2,
+            decimal=BOUNDING_BOX_PRECISION_TOLERANCE,
+        )
+
+
+def create_detections(detections_from_file: np.ndarray) -> detections_and_time.DetectionsAndTime:
+    """
+    Create DetectionsAndTime from expected detections.
+    Format: [confidence, label, x_1, y_1, x_2, y_2].
+    """
+    result, detections = detections_and_time.DetectionsAndTime.create(0)
+    assert result
+    assert detections is not None
+
+    if detections_from_file.size == 0:
+        return detections
+
+    if detections_from_file.ndim == 1:
+        detections_from_file = detections_from_file.reshape(1, -1)
+
+    assert (
+        detections_from_file.shape[1] == 6
+    ), f"Expected detections shape mismatch: {detections_from_file.shape}"
+
+    for detection_data in detections_from_file:
+        confidence, label, x_1, y_1, x_2, y_2 = detection_data
+        bounds = np.array([x_1, y_1, x_2, y_2])
+        result, detection = detections_and_time.Detection.create(
+            bounds=bounds,
+            label=int(label),
+            confidence=confidence,
```

Comment:
Eh... I don't like named arguments. But this is a nit so can merge without changing this.